### PR TITLE
Fix error where launching a game from command line with -p would erase it from scummvm.ini

### PR DESCRIPTION
**Pull request by rsf92**

### CODE REVIEW

File: common/str.cpp [hunk 1]
```diff
@@ -313,6 +313,11 @@ String String::vformat(const char *fmt, va_list args) {
 	return output;
 }
 
+//static
+String String::generateUniqueDomain( String gameId, int& suffixN) {
```
Comment (sev-) [inline common/str.cpp:317]: This does not belong here, please move it to `EngineManager`.

Also, code formatting is off

File: common/str.h [hunk 2]
```diff
@@ -195,6 +195,7 @@ class String : public BaseString<char> {
 	 */
 	static String vformat(const char *fmt, va_list args);
 
+	static String generateUniqueDomain( String gameId, int& suffixN);
```
Comment (sev-) [inline common/str.h:198]: Code formatting (besides moving it to a better place.

**Review by sev- — CHANGES_REQUESTED** [review]
THere are several issues with the current code

File: base/plugins.cpp [hunk 3]
```diff
@@ -765,13 +774,8 @@ Common::String EngineManager::createTargetForGame(const DetectedGame &game) {
 
 	assert(!domain.empty());
 	if (ConfMan.hasGameDomain(domain)) {
-		int suffixN = 1;
 		Common::String gameid(domain);
```
Comment (sev-) [inline base/plugins.cpp:777]: What is the purpose of this variable now?

Comment (rsf92) [reply base/plugins.cpp:777]: Well, i suppose i could just pass domain as well, yeah.

**Comment by sev-** [comment]
Thank you!

**Pull request merged**